ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py

```diff
@@ -0,0 +1,50 @@
+# global
+import numpy as np
+from hypothesis import given, strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy.functional.backends.tensorflow as ivy_tf
+
+
```

Comment:
The use of the  decorator handle_cmd_line_args has been changed a little. So now we're wrapping the  given decorator, instead of the test function. This is done so as to set the backend before assigning any values in the given decorator. We should now be using ivy.current_backend_str().valid_dtypes inside our helper functions, instead of ivy_np.valid_dtypes(i'm currently in the process of updating, this). Also, please don't pass in anything for as_variable, native_array, container, with_out, instance_method inside given, since that is being handled by the decorator, and it will only overwrite the values for these flags, which we don't want.

Reply:
I've made some of the changes you mentioned here but I don't think I'm fully understanding everything.

"We should now be using ivy.current_backend_str().valid_dtypes inside our helper functions, instead of ivy_np.valid_dtypes"

`ivy.current_backend_str()` this function returns `Union[str, None]` so therefore it cannot have an attribute of `.valid_dtypes`. Did you mean to mention another function else? Also is this to replace the use of `ivy_tf.valid_float_dtypes` on line 16?

The test itself is failing still, but I'm not totally sure why, I've look at recently merged tests as examples and can't seem to figure out what's wrong, perhaps I'm misusing the `@handle_cmd_line_args` function wrapper?

Thank you for your help 🙃.

Reply:
Well, I supposed it should be about ivy.current_backend() funcition